lib/types.nix

```diff
@@ -443,6 +443,13 @@ rec {
       functor = (defaultFunctor name) // { wrapped = [ t1 t2 ]; };
     };
 
+    # Any of the types in the given list
+    oneOf = ts:
+      let
+        head' = if ts == [] then throw "types.oneOf needs to get at least one type in its argument" else head ts;
+        tail' = tail ts;
+      in foldl either head' tail';
```

Comment:
Use `foldl'` here, lazy `foldl` is basically never what you want.

Reply:
I can’t believe we repeated the same mistake Haskell made …

Reply:
I thought about that, but I wasn't quite sure if it was really beneficial to use `foldl'` because the accumulator is a type (an attrset)

Reply:
`builtins.foldl'` is quite good at building attrsets